src/main/dynamo/QueryBuilder.ts

```diff
@@ -47,9 +47,14 @@ export class QueryBuilder<T extends TaggedModel> extends QueryExpressionBuilder<
   }
 
   async exec(): Promise<GroupedModels<T>> {
+    const query = this.createQueryInput({})
     const iterator = pagedIterator<DocumentClient.QueryInput, T>(
       {},
-      (options) => this.buildQuery(options),
```

Comment:
The root cause was this method was getting called on every page of results. But, it's not a pure function as it updates the attribute name/value map passed to Dynamo (which...is another thing that should be addressed)